mapie/tests/test_classification.py

```diff
@@ -1939,16 +1972,147 @@ def test_fit_parameters_passing() -> None:
+    with fit_params on the expected behavior of predict_params
+    Checks that conformity_scores from train and y_pred from test are 0.
+    """
+    X_train, X_test, y_train, y_test = (
+        train_test_split(X, y, test_size=0.2, random_state=random_state)
+    )
+    custom_gbc = CustomGradientBoostingClassifier(random_state=random_state)
+    score = LACConformityScore()
+    mapie_model = MapieClassifier(estimator=custom_gbc, conformity_score=score)
+    fit_params = {'monitor': early_stopping_monitor}
+    predict_params = {'check_predict_params': True}
+    mapie_model = mapie_model.fit(
+        X_train, y_train,
+        fit_params=fit_params,
+        predict_params=predict_params
+    )
+    y_pred = mapie_model.predict(X_test, agg_scores="mean", **predict_params)
+
+    expected_conformity_scores = np.ones((X_train.shape[0], 1))
+
+    np.testing.assert_allclose(mapie_model.conformity_scores_,
+                               expected_conformity_scores)
+    np.testing.assert_allclose(y_pred, 0)
+
+
+def test_using_one_predict_parameter_into_predict_but_not_in_fit() -> None:
+    """
+    Test that using predict parameters in the predict method
+    without using predict_parameter in the fit method raises an error.
+    """
+    custom_gbc = CustomGradientBoostingClassifier(random_state=random_state)
+    X_train, X_test, y_train, y_test = (
+        train_test_split(X, y, test_size=0.2, random_state=random_state)
+    )
+    mapie = MapieClassifier(estimator=custom_gbc)
+    predict_params = {'check_predict_params': True}
+    mapie_fitted = mapie.fit(X_train, y_train)
+
+    with pytest.raises(ValueError, match=(
+        fr".*Using 'predict_param' '{predict_params}' "
+        r"without using one 'predict_param' in the fit method\..*"
+        r"Please ensure a similar configuration of 'predict_param' "
+        r"is used in the fit method before calling it in predict\..*"
+    )):
+        mapie_fitted.predict(X_test, agg_scores="mean", **predict_params)
+
+
+def test_using_one_predict_parameter_into_fit_but_not_in_predict() -> None:
+    """
+    Test that using predict parameters in the fit method without using
+    predict_parameter in the predict method raises an error.
+    """
+    custom_gbc = CustomGradientBoostingClassifier(random_state=random_state)
+    X_train, X_test, y_train, y_test = (
+        train_test_split(X, y, test_size=0.2, random_state=random_state)
+    )
+    mapie = MapieClassifier(estimator=custom_gbc)
+    predict_params = {'check_predict_params': True}
+    mapie_fitted = mapie.fit(X_train, y_train, predict_params=predict_params)
+
+    with pytest.raises(ValueError, match=(
+        r"Using one 'predict_param' in the fit method "
+        r"without using one 'predict_param' in the predict method. "
+        r"Please ensure a similar configuration of 'predict_param' "
+        r"is used in the predict method as called in the fit."
+    )):
+        mapie_fitted.predict(X_test)
```

Comment:
Hey @thibaultcordier, why do you want to raise a warning. We believed an error was more appropriate, since if we don't use `predict_params` in both, it's clearly an issue.